### PR TITLE
Updating things

### PR DESCRIPTION
**Pull request by EiffL**
This pull request introduces a simulation budget scan framework to evaluate how the constraining power of learned summary statistics scales with the number of training simulations, particularly for the ResNet-18 compressor. It includes significant enhancements to the data pipeline, model configuration, and experiment tracking, as well as new scripts for launching, managing, and visualizing budget scan experiments.

**Key changes:**

### Experiment Configuration and Data Pipeline

* Added a new experiment config file `resnet18.yaml` specifying a ResNet-18 based model, using pretrained weights, cosine LR schedule, and all relevant trainer and logger settings for the budget scan.
* Updated `ChallengeDataModule` in `cosmoford/dataset.py` to accept a `max_train_samples` parameter, allowing the training set size to be limited for budget scan experiments. The `setup()` method now truncates the training dataset if this parameter is set, ensuring reproducible subsets. [[1]](diffhunk://#diff-ba64fc5857d8dbfe4f3e7be3071efa0c39f46a1faec690762819b52de2a249c4L72-R73) [[2]](diffhunk://#diff-ba64fc5857d8dbfe4f3e7be3071efa0c39f46a1faec690762819b52de2a249c4R83-R90) [[3]](diffhunk://#diff-ba64fc5857d8dbfe4f3e7be3071efa0c39f46a1faec690762819b52de2a249c4R165-R170)

### Model Enhancements

* Extended `RegressionModelNoPatch` in `cosmoford/models_nopatch.py` to support ResNet-18 as a backbone, including adaptation for 1-channel input and pretrained weights. Also added support for cosine learning rate scheduling, multi-noise validation averaging, and a new `compress` method for extracting summary representations. [[1]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dR10) [[2]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dR41-R77) [[3]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dR141-R144) [[4]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dR176-R206) [[5]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dR218-R227) [[6]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dL290-R361) [[7]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dL323-L330) [[8]](diffhunk://#diff-65a46435048607b7e219f269f2087b57f8f2343ef00ef4b79f110cc629d1de0dL339-R420)

### Experimentation and Visualization Scripts

* Added `scripts/plot_budget_scan.py`, a script to query wandb for runs tagged as budget scans, extract the best validation scores for each budget, and plot constraining power versus simulation budget.
* Added `scripts/plot_fom_budget.py`, a Modal-based script to aggregate NPE results from a shared volume, plot Figure of Merit (FoM) versus simulation budget, and save the plot as a PDF.

### Documentation

* Added a detailed plan in `docs/plans/2026-03-07-budget-scan-design.md` outlining the motivation, methodology, and implementation steps for the simulation budget scan.

### CODE REVIEW

File: scripts/run_npe_budget_scan.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    if compressor is None:
+        compressor = RegressionModelNoPatch.load_from_checkpoint(ckpt_path, map_location=device)
+        compressor.eval()
+        compressor.to(device)
+        for p in compressor.parameters():
+            p.requires_grad = False
+
+    # Load holdout kappa maps for FoM eval (need raw maps for fresh noisy obs)
+    print("Loading holdout maps for FoM evaluation...")
+    holdout = load_dataset("CosmoStat/neurips-wl-challenge-holdout", split="train")
```
Comment (Justinezgh) [inline scripts/run_npe_budget_scan.py]: `holdout = load_dataset("CosmoStat/neurips-wl-challenge-holdout", split="fiducial")`

**Review by Justinezgh — APPROVED** [review]
Just need to update the FoM fiducials. But then let's go to merge :)
